dialects/postgresql/src/main/kotlin/app/cash/sqldelight/dialects/postgresql/grammar/mixins/BindParameterMixin.kt

```diff
@@ -0,0 +1,12 @@
+package app.cash.sqldelight.dialects.postgresql.grammar.mixins
+
+import app.cash.sqldelight.dialect.grammar.mixins.BindParameterMixin
+import com.intellij.lang.ASTNode
+
+abstract class BindParameterMixin(node: ASTNode) : BindParameterMixin(node) {
+  override fun replaceWith(isAsync: Boolean, index: Int): String = when {
+    text == "DEFAULT" -> text
+    isAsync -> "$$index"
```

Comment:
this feels like a driver implementation detail, that specifically for R2DBC we need indexed parameters.

We're getting closer and I'm very happy to have functioning tests for this stuff now, so this is really the only thing I'm still hung up on

Reply:
Not only r2dbc. Postgres native needs indexed parameters too: https://github.com/hfhbd/postgres-native-sqldelight/blob/1a82ac52d6ab6246644fe8216932fd348afd1a9a/postgres-native-sqldelight-driver/src/commonMain/kotlin/app/softwork/sqldelight/postgresdriver/PostgresNativeDriver.kt#L165

Reply:
I kinda like that approach more, doing it at the driver level

Reply:
But at driver level you only have strings. So "SELECT '?'" or "LIKE 'foo?%'" does not work. (Except using sql-psi at driver level during runtime again before executing each stmt)

Reply:
alright, I buy that

last suggestion then: could we just _always_ replace arguments for the postgres driver with their indexed argument, even if its not the async driver? So that parameter can be removed and we're just saying for this dialect this is the expected parameter format

Reply:
What parameter do you want to drop?
index => How do you want to do this? Hardcode a check in sqldelight-core to replace the parameter wildcard with a specific string?
isAsync => I have to check it

Reply:
As expected, using `$1` does not work on JDBC and results in an org.postgresql.util.PSQLException, because you try to bind a parameter to an unknown one: 
```
In JDBC, the question mark (?) is the placeholder for the positional parameters of a PreparedStatement. There are, however, a number of PostgreSQL® operators that contain a question mark. To keep such question marks in an SQL statement from being interpreted as positional parameters, use two question marks ( ?? ) as escape sequence. You can also use this escape sequence in a Statement , but that is not required. Specifically only in a Statement a single ( ? ) can be used as an operator.
```
https://jdbc.postgresql.org/documentation/query/#using-the-statement-or-preparedstatement-interface

But I removes some useless check/binding implementations because I already checked for `DEFAULT`

Reply:
sounds good. I'm still wishing that this logic was in the driver somewhere instead of the dialect since it looks like a driver constraint, not a postgres one, but lets just merge to fix the underlying issue and then if it becomes a problem later we can revisit.